### PR TITLE
build: fix linking against wasm-c-api on macOS.

### PR DESCRIPTION
**Pull request by PiotrSikora**
While there, re-enable make/free accounting, since it was broken
for the same reason (implementation was leaked into headers).

Fixes envoyproxy/envoy-wasm#18.

Signed-off-by: Piotr Sikora <piotrsikora@google.com>

### CODE REVIEW

**Comment by PiotrSikora** [comment]
cc @bianpengyuan

File: ci/build_container/build_recipes/v8.sh [hunk 1]
```diff
@@ -53,29 +53,26 @@ cd ../..
...
  CC_FLAGS = ${C_FLAGS}
 -LD_FLAGS = -fsanitize-memory-track-origins -fsanitize-memory-use-after-dtor
 +LD_FLAGS =
 
  C_COMP = clang
+
 EOF
 
-# 3. Force full GC when destroying VMs.
+# 2. Force full GC when destroying VMs.
```
Comment (jplevyak) [inline ci/build_container/build_recipes/v8.sh]: Now that DEBUG is set can you revert the change to #if 1 ?

Comment (PiotrSikora) [reply ci/build_container/build_recipes/v8.sh]: Good catch, thanks! Done.
